test/integration/scheduler/scheduler_test.go

```diff
@@ -1234,32 +1237,46 @@ var _ = ginkgo.Describe("Scheduler", func() {
 				},
 			}
 			gomega.Expect(k8sClient.Create(ctx, matchingNS)).To(gomega.Succeed())
+
+			strictFIFOLocalQueue = testing.MakeLocalQueue("strict-fifo-q", matchingNS.Name).ClusterQueue(strictFIFOClusterQ.Name).Obj()
+			gomega.Expect(k8sClient.Create(ctx, strictFIFOLocalQueue)).Should(gomega.Succeed())
+
+			cqTeamA = testing.MakeClusterQueue("team-a").
+				ResourceGroup(*testing.MakeFlavorQuotas(onDemandFlavor.Name).Resource(corev1.ResourceCPU, "5").Obj()).
+				Cohort(chName).
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, cqTeamA)).Should(gomega.Succeed())
+
+			cqSharedResources = testing.MakeClusterQueue("shared-resources").
+				ResourceGroup(
+					*testing.MakeFlavorQuotas(onDemandFlavor.Name).Resource(corev1.ResourceCPU, "5").Obj()).
+				Cohort(chName).
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, cqSharedResources)).Should(gomega.Succeed())
 		})
 
 		ginkgo.AfterEach(func() {
 			gomega.Expect(util.DeleteNamespace(ctx, k8sClient, matchingNS)).To(gomega.Succeed())
 			gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
 			util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, strictFIFOClusterQ, true)
+			util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, cqTeamA, true)
+			util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, cqSharedResources, true)
 			util.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, onDemandFlavor, true)
 		})
 
 		ginkgo.It("Should schedule workloads by their priority strictly", func() {
-			strictFIFOQueue := testing.MakeLocalQueue("strict-fifo-q", matchingNS.Name).ClusterQueue(strictFIFOClusterQ.Name).Obj()
-
 			ginkgo.By("Creating workloads")
-			wl1 := testing.MakeWorkload("wl1", matchingNS.Name).Queue(strictFIFOQueue.
+			wl1 := testing.MakeWorkload("wl1", matchingNS.Name).Queue(strictFIFOLocalQueue.
 				Name).Request(corev1.ResourceCPU, "2").Priority(100).Obj()
 			gomega.Expect(k8sClient.Create(ctx, wl1)).Should(gomega.Succeed())
-			wl2 := testing.MakeWorkload("wl2", matchingNS.Name).Queue(strictFIFOQueue.
+			wl2 := testing.MakeWorkload("wl2", matchingNS.Name).Queue(strictFIFOLocalQueue.
 				Name).Request(corev1.ResourceCPU, "5").Priority(10).Obj()
 			gomega.Expect(k8sClient.Create(ctx, wl2)).Should(gomega.Succeed())
 			// wl3 can't be scheduled before wl2 even though there is enough quota.
-			wl3 := testing.MakeWorkload("wl3", matchingNS.Name).Queue(strictFIFOQueue.
+			wl3 := testing.MakeWorkload("wl3", matchingNS.Name).Queue(strictFIFOLocalQueue.
 				Name).Request(corev1.ResourceCPU, "1").Priority(1).Obj()
 			gomega.Expect(k8sClient.Create(ctx, wl3)).Should(gomega.Succeed())
 
-			gomega.Expect(k8sClient.Create(ctx, strictFIFOQueue)).Should(gomega.Succeed())
```

Comment:
The reason for creating this LocalQueue at this point is that we know that all the workloads exist and will be competing with each other, as opposed to arriving one by one.

Maybe we can change this to be more similar to #1544

Reply:
Updated